mail_sendgrid/models/sendgrid_template.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015-2017 Compassion CH (http://www.compassion.ch)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import models, fields, api, exceptions, _
+from odoo.tools.config import config
+
+import json
+import re
+import logging
+
+
+_logger = logging.getLogger(__name__)
+
+
+try:
+    import sendgrid
+except ImportError:
+    _logger.error("ImportError raised while loading module.")
+    _logger.debug("ImportError details:", exc_info=True)
+
+
+class SendgridTemplate(models.Model):
+    """ Reference to a template available on the SendGrid user account. """
+    _name = 'sendgrid.template'
+
+    ##########################################################################
+    #                                 FIELDS                                 #
+    ##########################################################################
+    name = fields.Char()
+    remote_id = fields.Char(readonly=True)
+    html_content = fields.Html(readonly=True)
+    plain_content = fields.Text(readonly=True)
+    detected_keywords = fields.Char(compute='_compute_keywords')
+
+    def _compute_keywords(self):
+        for template in self:
+            if template.html_content:
+                keywords = template.get_keywords()
+                self.detected_keywords = ';'.join(keywords)
+
+    @api.model
+    def update(self):
```

Comment:
:stop_sign: `update` is an[ internal method](https://github.com/odoo/odoo/blob/10.0/odoo/models.py#L5010) and is used to update in memory record w/ writing/creating things.